go/processor/floria/calls.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"fmt"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+func Call(interpreter tosca.Interpreter, transaction tosca.Transaction, context tosca.TransactionContext) (tosca.Result, error) {
+
+	blockParameters := tosca.BlockParameters{}
+
+	transactionParameters := tosca.TransactionParameters{
+		Origin:     transaction.Sender,
+		GasPrice:   transaction.GasPrice,
+		BlobHashes: []tosca.Hash{},
+	}
+
+	codeHash := context.GetCodeHash(*transaction.Recipient)
+	code := context.GetCode(*transaction.Recipient)
+
+	params := tosca.Parameters{
+		BlockParameters:       blockParameters,
+		TransactionParameters: transactionParameters,
+		//Context:               runContext, todo implement
+		Kind:      tosca.Call,
+		Static:    false,
+		Depth:     0, // todo add depth check
+		Gas:       transaction.GasLimit,
+		Recipient: *transaction.Recipient,
+		Sender:    transaction.Sender,
+		Input:     transaction.Input,
+		Value:     transaction.Value,
+		CodeHash:  &codeHash,
+		Code:      code,
+	}
+
+	snapshot := context.CreateSnapshot()
+	if err := transferValue(transaction, context); err != nil {
+		context.RestoreSnapshot(snapshot)
+		return tosca.Result{}, err
+	}
+
+	result, err := interpreter.Run(params)
+	if err != nil || !result.Success {
+		context.RestoreSnapshot(snapshot)
+	}
+
+	return result, err
+}
+
+func transferValue(transaction tosca.Transaction, context tosca.TransactionContext) error {
```

Comment:
if `transferValue` will be needed in multiple call functions, will they all happen here or should it be in a more neutral file where it can be reutilized? maybe this is a premature optimization.

Reply:
I think all of the call implementations should be in this file. Maybe we are even able to merge the different calls into a single function to reduce code duplication. The `transferValue` function should not be required out of this file.